security/src/main/java/no/difi/move/common/oauth/Oauth2JwtAccessTokenProvider.java

```diff
@@ -0,0 +1,54 @@
+package no.difi.move.common.oauth;
+
+import com.google.common.collect.Sets;
+import java.time.Instant;
+import java.util.Date;
+import org.springframework.security.oauth2.client.resource.OAuth2ProtectedResourceDetails;
+import org.springframework.security.oauth2.client.resource.UserApprovalRequiredException;
+import org.springframework.security.oauth2.client.resource.UserRedirectRequiredException;
+import org.springframework.security.oauth2.client.token.AccessTokenProvider;
+import org.springframework.security.oauth2.client.token.AccessTokenRequest;
+import org.springframework.security.oauth2.common.DefaultOAuth2AccessToken;
+import org.springframework.security.oauth2.common.OAuth2AccessToken;
+import org.springframework.security.oauth2.common.OAuth2RefreshToken;
+
+/**
+ *
+ * @author Nikolai Luthman <nikolai dot luthman at inmeta dot no>
+ */
+public class Oauth2JwtAccessTokenProvider implements AccessTokenProvider {
+
+    private final NimbusDSJwtTokenClient tokenClient;
+
+    public Oauth2JwtAccessTokenProvider(NimbusDSJwtTokenClient tokenClient) {
+        this.tokenClient = tokenClient;
+    }
+
+    private DefaultOAuth2AccessToken getAccessToken() {
+        JwtTokenResponse oidcTokenResponse = tokenClient.fetchToken();
+        DefaultOAuth2AccessToken oa2at = new DefaultOAuth2AccessToken(oidcTokenResponse.getAccessToken());
+        oa2at.setExpiration(Date.from(Instant.now().plusSeconds(250)));
+        oa2at.setScope(Sets.newHashSet(oidcTokenResponse.getScope()));
+        return oa2at;
+    }
+
+    @Override
+    public OAuth2AccessToken obtainAccessToken(OAuth2ProtectedResourceDetails details, AccessTokenRequest parameters) throws UserRedirectRequiredException, UserApprovalRequiredException {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the declaration of thrown exception 'org.springframework.security.oauth2.client.resource.UserRedirectRequiredException' which is a runtime exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the declaration of thrown exception 'org.springframework.security.oauth2.client.resource.UserApprovalRequiredException' which is a runtime exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)